### PR TITLE
Install one pip binary only (not two)

### LINKED ISSUE #133: python-pip and pip issues with salt/docker
The compose.sls state uses `pkg.installed` for **python-pip**, and `pip.installed` for **pip**.   Do both need to be handled?
```
compose-pip:
  pkg.installed:
    - name: python-pip
  pip.installed:
    - name: pip
    - upgrade: True
```
After OS installation the `python-pip` package will typically be installed, but pip maybe missing (on production environments at least).  And some docker modules might be installed already in  `/usr/lib/python ...` because it maybe distro default.
```
$ pip list | grep docker
docker-pycreds (0.2.2)
dockerpty (0.4.1)

$ ls /usr/lib/python2.7/dist-packages/docker
dockerpycreds/                  docker_pycreds-0.2.2.dist-info/
```

After `docker.compose.compose-pip` state, pip installs to `/usr/local/lib` by default (i.e. pip's pip).
```
$ pip list | grep docker
docker (2.7.0)
docker-compose (1.19.0)
docker-pycreds (0.2.2)
dockerpty (0.4.1)
```

This causes [weird issues](https://github.com/GoogleCloudPlatform/google-cloud-python/issues/4352#issuecomment-342561086) 
```
[ERROR   ] State 'docker_image.present' was not found in SLS 'docker.compose-ng'
Reason: 'docker_image' __virtual__ returned False: 'docker.version' is not available.

[ERROR   ] State 'docker_image.present' was not found in SLS 'docker.compose-ng'
Reason: 'docker_image' __virtual__ returned False: 'docker.version' is not available.

[ERROR   ] State 'docker_image.present' was not found in SLS 'docker.compose-ng'
Reason: 'docker_image' __virtual__ returned False: 'docker.version' is not available.
```

Possibly other weird stuff too.
```
^[[33mWARNING:^[[0m Dependency conflict: an older version of the 'docker-py' package may be polluting the namespace. If you're experiencing crashes, run the following command to remedy the issue:
pip uninstall docker-py; pip uninstall docker; pip install docker
[WARNING ] /usr/local/lib/python2.7/dist-packages/docker/api/client.py:163: UserWarning: The minimum API version supported is 1.21, but you are using version 1.18. It is recommended you either upgrade Docker Engine or use an older version of Docker SDK for Python.
  'Python.'.format(MINIMUM_DOCKER_API_VERSION, self._version)
```
And more ....

```
>>> import docker
>>> from docker import pulled
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name pulled
```

 **Finding the happy path takes time, and customizations, etc***
```
# sudo salt-call --local cmd.run 'python -c "import docker ; import docker.utils; print(docker.version); help(docker)"'
local:
    3.0.1
```
This issue can also be replicated with `packages-formula` alone, or with `docker-formula`.  I know this issue is a known hazard, but flagging it because I saw this.

Thanks.

### PR DESCRIPTION
**Pull request by noelmcloughlin**
The formula should install one **pip implementation** only (not two).  
Resolves #133, avoids #144 & #96

### CODE REVIEW

File: .kitchen.yml [hunk 1]
```diff
@@ -15,14 +15,33 @@ provisioner:
...
+  - name: ubuntu-16.04
+    driver_config:
+      provision_command:
+        - apt-get update
+      run_command: /lib/systemd/systemd
   - name: debian-jessie
     driver_config:
+      provision_command:
+        - apt-get update
       run_command: /sbin/init
```
Comment (javierbertoli) [inline .kitchen.yml]: Mind you change this `run_command` to `/lib/systemd/systemd`, for consistency?

Comment (noelmcloughlin) [reply .kitchen.yml]: Sure, good catch @javierbertoli

**Comment by javierbertoli** [comment]
@noelmcloughlin can you fix the conflicting files so we can merge?

**Comment by javierbertoli** [comment]
Thanks, @noelmcloughlin!

**Pull request merged**